core/templates/dev/head/domain/exploration/SimpleEditorQuestionsDataService.js

```diff
@@ -0,0 +1,150 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for storing information about the current exploration
+ *   in the form of question fields.
+ */
+
+oppia.factory('SimpleEditorQuestionsDataService', [
+  'explorationData', 'explorationInitStateNameService',
+  'explorationStatesService',
+  function(
+      explorationData, explorationInitStateNameService,
+      explorationStatesService) {
+    var questions = null;
+
+    var SUPPORTED_INTERACTION_TYPES = [{
+      id: 'MultipleChoiceInput',
+      name: 'Multiple choice'
+    }];
+
+    var allowedInteractionIds = SUPPORTED_INTERACTION_TYPES.map(
+      function(interactionData) {
+        return interactionData.id;
+      }
+    );
+
+    // Attempts to convert the exploration into a set of fields. If this is
+    // successful, returns the extracted fields, otherwise returns null.
+    var getQuestions = function() {
+      var stateNamesInOrder = [];
+      var currentStateName = explorationInitStateNameService.savedMemento;
+      var questionData = [];
+
+      var iterations = 0;
+      while (currentStateName) {
+        iterations++;
+        if (iterations > 100) {
+          console.error('Too many iterations in while loop');
+          break;
+        }
+        if (stateNamesInOrder.indexOf(currentStateName) !== -1) {
+          // There is a cycle in the exploration graph.
+          return null;
+        }
+        stateNamesInOrder.push(currentStateName);
+
+        var stateData = explorationStatesService.getState(currentStateName);
+        var interactionId = stateData.interaction.id;
+        if (!interactionId) {
+          break;
+        }
+
+        if (allowedInteractionIds.indexOf(interactionId) === -1) {
+          // The interaction for this state is not supported in the simple
+          // editor.
+          return null;
+        }
+
+        // Is the default answer group a self-loop, and is there exactly one
+        // non-self-loop destination among the non-default answer groups, and
+        // are there no fallbacks or param changes?
+        // TODO(sll): This needs to be generalized into a per-interaction
+        // validity check, that also includes checks for the customization
+        // args.
+        var destinationStateNames = [];
+        stateData.interaction.answer_groups.forEach(function(group) {
+          if (group.outcome.dest !== currentStateName) {
+            destinationStateNames.push(group.outcome.dest);
+          }
+        });
+
+        var defaultOutcome = stateData.interaction.default_outcome;
+        if (destinationStateNames.length > 1 ||
+            stateData.param_changes.length > 0 ||
+            defaultOutcome.dest !== currentStateName ||
+            defaultOutcome.param_changes.length > 0 ||
+            stateData.interaction.fallbacks.length > 0) {
```

Comment:
This whole function is going to be redone later; some checks that need to be in place are not in place yet. So, punting this for later.